data_model/src/account.rs

```diff
@@ -148,24 +147,28 @@ pub mod model {
     /// Condition which checks if the account has the right signatures.
     #[derive(
         Debug,
+        Default,
         Display,
         Clone,
         PartialEq,
         Eq,
         PartialOrd,
         Ord,
-        Constructor,
         Decode,
         Encode,
         Deserialize,
         Serialize,
         IntoSchema,
     )]
-    #[serde(transparent)]
-    #[repr(transparent)]
-    // SAFETY: `SignatureCheckCondition` has no trap representation in `EvalueatesTo<bool>`
-    #[ffi_type(unsafe {robust})]
-    pub struct SignatureCheckCondition(pub EvaluatesTo<bool>);
+    #[ffi_type(opaque)]
+    #[allow(clippy::enum_variant_names)]
+    pub enum SignatureCheckCondition {
+        #[default]
+        AnyAccountSignatures,
+        AllAccountSignatures,
+        #[display(fmt = "[{_0:?}]")]
+        AllCustomSignatures(Vec<PublicKey>),
```

Comment:
I believe this `Vec` is immutable. Could we expand #2573 to support other immutable `Vec` types not only bytes?

Reply:
Yeah, I think so

Reply:
I've pushed these changes to #3854 that generalize this. Would either need to wait for it to merge or use some git-fu..